content/guides/10-minute-tutorial.md

```diff
@@ -792,12 +793,281 @@ Feature: Is it Friday yet?
+
+Update the `is-it-friday-yet.feature` file.
+```gherkin
+Feature: Is it Friday yet?
+  Everybody wants to know when it's Friday
+
+  Scenario: Sunday isn't Friday
+    Given today is Sunday
+    When I ask whether it's Friday yet
+    Then I should be told "Nope"
+
+  Scenario: Friday is Friday
+    Given today is Friday
+    When I ask whether it's Friday yet
+    Then I should be told "TGIF"
+```
+
+{{% block "javascript" %}}
+Update the `stepdefs.js` file.
+```javascript
+const assert = require('assert');
+const { Given, When, Then } = require('cucumber');
+
+function isItFriday(today) {
+  return (today === "Friday") ? "TGIF" : "Nope";
+}
+
+// These two Given functions are very similar but, as written, require separate functions
+Given('today is Sunday', function () {
+  this.today = 'Sunday';
+});
+Given('today is Friday', function () {
+  this.today = 'Friday';
+});
+
+// This is a shared function and should be moved to a shared.js file
+When('I ask whether it\'s Friday yet', function () {
+  this.actualAnswer = isItFriday(this.today);
+});
+
+// This is also a shared function that uses a variable, {string}
+// This should also be moved to a shared.js file
+Then('I should be told {string}', function (expectedAnswer) {
+  assert.equal(this.actualAnswer, expectedAnswer);
+});
+```
+{{% /block %}}
+
+{{% block "ruby" %}}
+```ruby
+module FridayStepHelper
+  def is_it_friday?(day)
+    day == 'Friday' ? 'TGIF' : 'Nope'
+  end
+end
+World FridayStepHelper
+
+Given("today is Sunday") do
+  @today = 'Sunday'
+end
+
+Given("today is Friday") do
+  @today = 'Friday'
+end
+
+# Then is a shared function
+When("I ask whether it's Friday yet") do
+  @actual_answer = is_it_friday?(@today)
+end
+
+# Then is a shared function
+Then("I should be told {string}") do |expected_answer|
+  expect(@actual_answer).to eq(expected_answer)
+end
+```
+{{% /block %}}
+
+Run Cucumber again:
+
+{{% block "javascript" %}}
+```shell
+......
+2 scenarios (2 passed)
+6 steps (6 passed)
+0m00.002s
+```
+{{% /block %}}
+
+{{% block "ruby" %}}
+```shell
+Feature: Is it Friday yet?
+  Everybody wants to know when it's Friday
+
+  Scenario: Sunday isn't Friday        # features/is_it_friday_yet.feature:4
+    Given today is Sunday              # features/step_definitions/stepdefs.rb:8
+    When I ask whether it's Friday yet # features/step_definitions/stepdefs.rb:17
+    Then I should be told "Nope"       # features/step_definitions/stepdefs.rb:22
+
+  Scenario: Friday is Friday           # features/is_it_friday_yet.feature:9
+    Given today is Friday              # features/step_definitions/stepdefs.rb:12
+    When I ask whether it's Friday yet # features/step_definitions/stepdefs.rb:17
+    Then I should be told "TGIF"       # features/step_definitions/stepdefs.rb:22
+
+2 scenarios (2 passed)
+6 steps (6 passed)
+0m0.040s
+```
+{{% /block %}}
+
```

Comment:
Add Java:

- If we write a *failing* test first (as per my suggestion), the result will be:
```
    Given today is "Sunday"            # Stepdefs.today_is(String)
    When I ask whether is's Friday yet # Stepdefs.i_ask_whether_is_s_Friday_yet()
    Then I should be told "Nope"       # Stepdefs.i_should_be_told(String)

  Scenario: Friday is Friday           # hellocucumber/is_it_friday.feature:9
    Given today is "Friday"            # Stepdefs.today_is(String)
    When I ask whether is's Friday yet # Stepdefs.i_ask_whether_is_s_Friday_yet()
    Then I should be told "TGIF"       # Stepdefs.i_should_be_told(String)
      org.junit.ComparisonFailure: expected:<[TGIF]> but was:<[Nope]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at hellocucumber.Stepdefs.i_should_be_told(Stepdefs.java:26)
	at ✽.I should be told "TGIF"(hellocucumber/is_it_friday.feature:12)


org.junit.ComparisonFailure: 
Expected :TGIF
Actual   :Nope
 <Click to see difference>


	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at hellocucumber.Stepdefs.i_should_be_told(Stepdefs.java:26)
	at ✽.I should be told "TGIF"(hellocucumber/is_it_friday.feature:12)
```
Note: mark as java block.

- For a passing test the result will be:
```
Feature: Is it Friday yet?
  Everybody wants to know when it's Friday

  Scenario: Sunday isn't Friday        # hellocucumber/is_it_friday.feature:4
    Given today is "Sunday"            # Stepdefs.today_is(String)
    When I ask whether is's Friday yet # Stepdefs.i_ask_whether_is_s_Friday_yet()
    Then I should be told "Nope"       # Stepdefs.i_should_be_told(String)

  Scenario: Friday is Friday           # hellocucumber/is_it_friday.feature:9
    Given today is "Friday"            # Stepdefs.today_is(String)
    When I ask whether is's Friday yet # Stepdefs.i_ask_whether_is_s_Friday_yet()
    Then I should be told "TGIF"       # Stepdefs.i_should_be_told(String)
```
Note: mark as java block.